sematic/db/queries.py

```diff
@@ -159,20 +158,23 @@ def get_run_status_details(
     """
     with db().get_session() as session:
         query_results = (
-            session.query(Run.id, Run.future_state, Run.external_jobs_json)
-            .filter(Run.id.in_(run_ids))
+            session.query(Run.id, Run.future_state).filter(Run.id.in_(run_ids)).all()
+        )
+        jobs = list(
```

Comment:
Why not do one join query instead?

Reply:
Ideally the query would look something like this:
```python
            session.query(Run.id, Run.future_state, sqlalchemy.func.array_agg(Job))
            .filter(Run.id.in_(run_ids))
            .filter(Job.kind == JobKind.run)
            .filter(Job.run_id == Run.id)
            .group_by(Run.id)
            .all()
```

Unfortunately I have been unable to get array_agg to work getting the whole Job object, which is what we'd need. We can get it to one query if we don't do the group by, but then we have to do the group-by ourselves. That said, it's not too computationally intensive to do so. Changed.